internals/secrethub/credential_disable_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/secrethub/secrethub-go/pkg/secrethub/fakeclient"
+)
+
+func TestCredentialDisableCommand_Run(t *testing.T) {
+
+	var testErr = errors.New("test")
+	const (
+		validFingerprint = "62542d734d7f3627"
+		warningMessage   = "A disabled credential can no longer be used to access SecretHub. " +
+			"This process can currently not be reversed.\n"
+	)
+	testCases := map[string]struct {
+		cmd          CredentialDisableCommand
+		force        bool
+		promptOut    string
+		out          string
+		in           string
+		disableErr   error
+		newClientErr error
+		promptErr    error
+		err          error
+	}{
+		"fail force no credential": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "",
+				force:       true,
+			},
+			err: errors.New("fingerprint argument must be set when using --force"),
+		},
+		"fail prompt error": {
+			promptErr: testErr,
+			err:       testErr,
+		},
+		"fail empty credential 3 times": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "",
+				force:       false,
+			},
+			promptOut: "What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n" +
+				"Please try again.\n" +
+				"What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n" +
+				"Please try again.\n" +
+				"What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n",
+			err: api.ErrInvalidFingerprint,
+		},
+		"fail abort": {
+			cmd: CredentialDisableCommand{
+				fingerprint: validFingerprint,
+				force:       false,
+			},
+			promptOut: fmt.Sprintf("Are you sure you want to disable the credential with fingerprint %s? [y/N]: ", validFingerprint),
+			in:        "n",
+			out:       warningMessage + "Aborting.\n",
+		},
+		"fail client error": {
+			newClientErr: testErr,
+			err:          testErr,
+		},
+		"fail force wrong credential": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "BillyBoy",
+				force:       true,
+			},
+			err: api.ErrInvalidFingerprint,
+		},
+		"fail too short fingerprint": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "6254",
+				force:       true,
+			},
+			err: api.ErrTooShortFingerprint,
+		},
+		"succeed force": {
+			cmd: CredentialDisableCommand{
+				fingerprint: validFingerprint,
+				force:       true,
+			},
+			out: warningMessage + "Credential disabled.\n",
+		},
+		"succeed no force": {
+			cmd: CredentialDisableCommand{
+				fingerprint: validFingerprint,
+				force:       false,
+			},
+			promptOut: fmt.Sprintf("Are you sure you want to disable the credential with fingerprint %s? [y/N]: ", validFingerprint),
+			in:        "y",
+			out:       warningMessage + "Credential disabled.\n",
+		},
+		"fail disable error": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3628",
```

Comment:
Why is this different from the `validFingerprint`?

Reply:
Because valid fingerprint is supposed to not return any error, while this one returns an error when the command tries to make a call to disable it.

Reply:
What would be the second question? :)

Reply:
It is different because although it is a valid fingerprint (it can be used as a short version of a credential fingerprint), it is not the one that will be disabled since it doesn't exist.

Reply:
The other comment, but I figured it out after submitting the review :sweat_smile: